cmd/jaeger/internal/integration/e2e_integration.go

```diff
@@ -109,7 +110,24 @@ func (s *E2EStorageIntegration) e2eInitialize(t *testing.T, storage string) {
 			return false
 		}
 		defer resp.Body.Close()
-		return resp.StatusCode == http.StatusOK
+		if s.HealthCheckEndpoint == "" {
```

Comment:
this is never true, see L96

On a separate question, is `ports.QueryHTTP` what you want to be checking (for non-Kafka tests)?

Reply:
ports.QueryHTTP was already written in the code before by Harshit.
Didn't bother to tinker with that.

Reply:
L96 is initializing the healthCheckEndpoint, but the s.HealthCheckEndpoint if "" still remains empty even after healthCheckEndpoint is initialized if I am right?

Reply:
what is the possible execution path for this variable to remain empty after this code?
```
	healthCheckEndpoint := s.HealthCheckEndpoint
	if healthCheckEndpoint == "" {
		healthCheckEndpoint = fmt.Sprintf("http://localhost:%d/", ports.QueryHTTP)
	}
```

Reply:
healthCheckEndpoint will not be empty.
But, the condition tests for if s.HealthCheckEndpoint is empty, and s.HealthCheckEndpoint will always be empty from where it is initialized.